Assets/SceneChangeDoor.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.SceneManagement;
+
+public class SceneChangeDoor : MonoBehaviour
+{
+    public GameObject player;
+    public string targetScene;
+    private bool playerInInteractionArea = false;
+    private InputControls input = null;
+
+    private void Awake()
+    {
+        input = new InputControls();
+        input.Enable();
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    private void OnTriggerEnter(Collider other)
+    {
+        Debug.Log($"Door interaction zone entered by {other.name}");
+        if (other.gameObject == player)
+        {
+            playerInInteractionArea = true;
+        }
+    }
+    private void OnTriggerExit(Collider other)
+    {
+        Debug.Log($"Door interaction zone exited by {other.name}");
+        if (other.gameObject == player)
+        {
+            playerInInteractionArea = false;
+
+        }
+    }
+
+#pragma warning disable IDE0051 // This is used by the PlayerInput component
+    private void OnInteract() {
+        Debug.Log("Interacting");
+        if (playerInInteractionArea)
+        {
+            Debug.Log($"Should change scenes to {targetScene}");
+            SceneManager.LoadScene(targetScene);
```

Comment:
Not needed for this PR but we should consider how we want to persist state when switching Scenes. Maybe this is already figured out by @sllewely 's load / saving? My guess is that this would currently wipe any progress made in the dungeon

Reply:
Yeah, probably a good idea to at least save before changing the scene